backend/src/index.ts

```diff
@@ -126,6 +131,57 @@ app.post(
   }
 );
 
+app.post("/auth/otp/generate", async(req: Request, res: Response) => {
+  try {
+    const { email } : { email: string} = req.body;
+    
+    if(!email) {
+      throw new Error("Email address expected.");
+    }
+
+    const token = await generateOTP(email, SALT_ROUNDS); 
+
+    if (token) {
+      await redisClient.set(email, token, { EX: 60 });
+      return res.status(200).json({ message: "ok" });
+    } else {
+      return res.status(400).json( {message: "Unexpected error while generating OTP."} );
+    }
+
+  } catch(error) {
+    return res.status(400).json({ message: (error as Error).message });
+  }
+});
+
+app.post("/auth/otp/verify", async(req: Request, res: Response) => {

```

Comment:
just a thought: should we combine the requests for otp verification and password changing as one? how could we securely implement a 'change password' route while ensuring that the provided otp has been verified beforehand?